### PR TITLE
Require WordPress 6.9 and WooCommerce 10.9

### PR DESCRIPTION
**Pull request by NeosinneR**
Closes PIN4WOO-116

### Changes proposed in this Pull Request:

* Require WordPress 6.9 and WooCommerce 10.9 in metadata and existing runtime checks.
* Pin a minimum-version test lane and let the installer select WooCommerce 10.9.0.
* Align requirement documentation and test setup; remove obsolete WC Admin test initialization.

### Detailed test instructions:

1. On an isolated site with PHP 7.4, WordPress 6.9, and WooCommerce 10.9.0, activate the built plugin and confirm there is no dependency warning.
2. Open the Pinterest settings or onboarding screen; confirm it loads without a dependency warning.
3. Deactivate the plugin, switch to WordPress 6.8, and attempt activation; WordPress must report that WordPress 6.9 is required.
4. Restore WordPress 6.9 and activate the plugin, then switch WooCommerce to 10.8.0; confirm its dependency notice requires WooCommerce 10.9 and plugin initialization stops. Restore WooCommerce 10.9.0 and confirm initialization resumes.

No feature flag is required.

### Additional details:

The minimum-version suite passed on PHP 7.4. All four WordPress/WooCommerce boundary cases pass on PHP 8.3; restoring the old WooCommerce minimum makes its below-minimum case fail. Installer selection, changed-line PHPCS, and real activation/dependency checks passed. The boundary test file passes full-file PHPCS; existing production-file PHPCS errors remain.

* [x] Auto-assign milestone

### Changelog entry

> Tweak - Raised minimum requirements to WordPress 6.9 and WooCommerce 10.9.

### CODE REVIEW

File: class-pinterest-for-woocommerce.php [hunk 1]
```diff
@@ -75,8 +75,8 @@ final class Pinterest_For_Woocommerce {
 		 */
 		const PLUGIN_REQUIREMENTS = array(
 			'php_version'      => '7.4',
-			'wp_version'       => '5.6',
-			'wc_version'       => '7.0',
+			'wp_version'       => '6.9',
+			'wc_version'       => '10.9.0',
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply class-pinterest-for-woocommerce.php:79]: Fixed in [c28af92f](https://github.com/woocommerce/pinterest-for-woocommerce/commit/c28af92f74161c7e123d518d01394c6e5dc3e145). Added cases for WooCommerce 10.8.99 (rejected) and 10.9.0 (accepted). Each runs the actual requirements check in a fresh PHP process so `WC_VERSION` can be set independently of the test bootstrap. All four WordPress/WooCommerce boundary cases and all five CI checks pass.

File: tests/Unit/PluginRequirementsTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	 * @param bool   $expected Whether the version is supported.
+	 */
+	public function test_wordpress_version_requirement( $version, $expected ) {
+		global $wp_version;
+
+		$original_version = $wp_version;
+		$wp_version       = $version; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited -- Exercise the runtime version check.
+
+		try {
+			$this->assertSame( $expected, Pinterest_For_Woocommerce::instance()->check_plugin_requirements() );
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply tests/Unit/PluginRequirementsTest.php:33]: Covered by the same fix in [c28af92f](https://github.com/woocommerce/pinterest-for-woocommerce/commit/c28af92f74161c7e123d518d01394c6e5dc3e145). I also restored the old WooCommerce 7.0 minimum in a separate test copy and confirmed that the new below-minimum case fails, so the test catches this regression.
